tests/functional/m/modified_iterating.py

```diff
@@ -93,3 +93,15 @@ def update_existing_key():
     for key in my_dict:
         new_key = key.lower()
         my_dict[new_key] = 1  # [modified-iterating-dict]
+
+
+class MyClass:  # pylint: disable=too-few-public-methods
+    """Regression test for https://github.com/PyCQA/pylint/issues/7380"""
+
+    def __init__(self) -> None:
+        self.attribute = [1, 2, 3]
+
+    def my_method(self):
+        """This should raise as we are deleting."""
+        for var in self.attribute:
+            del var  # [modified-iterating-list]
```

Comment:
```suggestion
    def modifying_list_(self):
    # should be fine to modify a copy
            for var in self.iterated_attributes.copy():
            del var  # [modified-iterating-list]
     # should raise a message when modifying
        for var in self.iterated_attributes:
            del var  # [modified-iterating-list]

```

Also check the false positive case. Fix the `.txt` file accordingly.

Reply:
@DanielNoord  yeah i re-read it and understood that the case which is not working is the class case. 
I don't fully understand the `append` example, how can you put `list.append` in a for loop?

Reply:
```python
for x in a_list:
    a_list.append(x)
```

The issue here is that `append` is an `Attribute` node on `a_list` just as `attribute` is an `Attribute` node in the following example:
```python
class MyClass:
    def __init__():
        self.attribute = [1,2,3]

    def method(self):
        for x in self.attribute:
            del x
```

`Astroid` only recognises that they are nodes that come after a period, not that they are attributes in the traditional sense of the word.
This complicates the logic we're currently using in the checker. I tried to fix it, but it turned out to be a pretty large refactor which probably should be taken in multiple steps for ease of reviewing.
To make sure we no longer crash I wanted to push this fix at least.

Reply:
You mean like when iterating through a dictionary `d` :  iterating on  `d.keys` is an  `Attribute` which is actually a fix the our error of iterating through `d`? something like that? so your current solution will throw false-postive now?

Reply:
I think you'll understand what you mean when you add the example I gave in the opening post to the functional tests. If you run `pytest` then you'll see that some of the `isinstance(node, nodes.Attribute)` test you are using to pick up `d.append` will now also pick up `MyClass.attribute`. This causes some issues and methods such as `_modified_iterating_list_cond` will need to be changed to allow checking both type of nodes.